singularity-eos/eos/modifiers/eos_unitsystem.hpp

```diff
@@ -35,9 +35,9 @@ using namespace eos_base;
 
 // tag dispatch for constructors for UnitSystem
 namespace eos_units_init {
-static struct ThermalUnitsInit {
+[[maybe_unused]] static struct ThermalUnitsInit {
 } thermal_units_init_tag;
-static struct LengthTimeUnitsInit {
+[[maybe_unused]] static struct LengthTimeUnitsInit {
 } length_time_units_init_tag;
```

Comment:
clang didn't care that these tags might be unused, maybe because it could tell they were for tag dispatch. But gcc did care. `[maybe_unused]` is a C++17 feature, so we can safely use it and assume it's compiler agnostic.